src/monaco.contribution.ts

```diff
@@ -46,21 +57,46 @@ export class LanguageServiceDefaultsImpl implements monaco.languages.typescript.
 		}
 
 		if (this._extraLibs[filePath]) {
-			throw new Error(`${filePath} already a extra lib`);
+			this._extraLibs[filePath].version++;

```

Comment:
Check if the content is the same via === before incrementing the version?

```suggestion
			if (this._extraLibs[filePath].content !== content) {
				this._extraLibs[filePath].version++;
				this._extraLibs[filePath].content = content;			 
			}
```